c_api/impl/io_c.h

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+// -*- c++ -*-
+
+#ifndef FAISS_IO_C_H
+#define FAISS_IO_C_H
+
+#include <faiss/impl/io.h>
+#include "../faiss_c.h"
+
+using faiss::IOReader;
```

Comment:
The `using` primitives and the struct inheritance below are not compatible with C, so I don't see how this is going to work. If anything, `CustomIOReader` and `CustomIOWriter` should be kept as private definitions in `.cpp` code, which would still allow for the reinterpret cast between the C and C++ versions.

Reply:
Makes sense, added a commit for this